PowerEditor/installer/nppSetup.nsi

```diff
@@ -178,16 +178,28 @@ updaterDone:
 	Call SetRoughEstimation		; This is rough estimation of files present in function copyCommonFiles
 	InitPluginsDir			; Initializes the plug-ins dir ($PLUGINSDIR) if not already initialized.
 	Call checkCompatibility		; check unsupported OSes and CPUs
-		
+	
 	; look for previously selected language
 	ClearErrors
 	Var /GLOBAL tempLng
 	ReadRegStr $tempLng HKLM "SOFTWARE\${APPNAME}" 'InstallerLanguage'
 	${IfNot} ${Errors}
-		StrCpy $LANGUAGE "$tempLng" ; set default language
+		${If} "$tempLng" == "Installer Language" ; check for the possible previous inappropriate value (from some of the 'Afrikaans' issues...)
+			WriteRegStr HKLM "SOFTWARE\${APPNAME}" 'InstallerLanguage' '${LANG_ENGLISH}' ; bad, so repair that with our factory default
+			StrCpy $LANGUAGE "${LANG_ENGLISH}" ; and preset accordingly to the above
+		${Else}
+			StrCpy $LANGUAGE "$tempLng" ; ok, so offer that previously used language as the installer default
+		${EndIf}
 	${EndIf}
 	
 	!insertmacro MUI_LANGDLL_DISPLAY
+	
+	; check for the possible NSIS lang-engine fault
+	; - some users (maybe with some Windows server editions?) reported a complete omission of the installer language-selection dlg
+	;   (the Welcome Screen is then shown directly instead and the $LANGUAGE var is filled up with the "Installer Language" nonsense)
+	${If} "$LANGUAGE" == "Installer Language"
+		StrCpy $LANGUAGE "${LANG_ENGLISH}" ; a NSIS inherent failure(?!), at least repair that to our factory default
+	${EndIf}

```

Comment:
I know the above 2 parts are the conclusion of test from you and @MalakymR.
However, I'm wondering if it's necessary.

As you said, current installer has 2 problems with iceland windows:
1. Language Selector appear, but detection failure so it's always Afrikaans which is the first choice.
2. Language Selector doesn't appear, the installer language is in Afrikaans during the installation.

Both problem should solve (partially, with acceptable solution) with my binary as I mention here:
https://github.com/notepad-plus-plus/notepad-plus-plus/pull/14215/files#r1349304416

So with it, we could have 2 situation:
1. Language Selector appear, but detection failure so it's always English which is the first choice.
2. Language Selector doesn't appear, the installer language is in English during the installation.

Could you confirm the above description?

Reply:
> Could you confirm the above description?

No, that is not correct.

> As you said, current installer has 2 problems with iceland windows:
> 
>     1. Language Selector appear, but detection failure so it's always Afrikaans which is the first choice.
> 
>     2. Language Selector doesn't appear, the installer language is in Afrikaans during the installation.

No. As I tried to explain [here](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/3844#issuecomment-1748610142), we have 2 independent types of the 'Afrikaans' problem:

A) Lang-selection dlg is shown but currently with invalid Afrikaans
- it is caused either by a specific Windows locale like Icelandic or by a previous invalid N++ lang-regkey value
- this one is easy to fix by that "English-1st-in-our-static-lang-list"

B) Lang-selection dlg is not shown at all (directly the  "Welcome Screen" with invalid Afrikaans)
- this one is NOT caused by a specific Windows locale like Icelandic and I cannot simulate this type of the NSIS error myself
- this is what @MalakymR has on his computer with [English (United Kingdom) locale](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/3844#issuecomment-1711349415)
- it is the worse (not fully fixable) one, here even a previous correct lang-regkey is ignored totally and $LANGUAGE installer variable is overwritten at runtime to the nonsense value "Installer Language" by NSIS itself (then this was consequently also written by us to the lang-regkey later because of we could not previously know that something like that is even possible...)

Reply:
What do you mean by "invalid Afrikaans" ?
Do you mean language detection failure so it's Afrikaans set as default?

Reply:
> A) Lang-selection dlg is shown but currently with invalid Afrikaans

So A has been fixed by setting English at the 1st position right?

Reply:
> B) Lang-selection dlg is not shown at all (directly the "Welcome Screen" with invalid Afrikaans)

B is not fixed completely, because  Lang-selection dlg is not shown at all, but "Welcome Screen" with English, thanks to the fix of setting English at the 1st position, right ?

Reply:
> B is not fixed completely, because Lang-selection dlg is not shown at all, but "Welcome Screen" with English, thanks to the fix of setting English at the 1st position, right ?

Yes and no.
Even when that "Welcome Screen" is then in English due to that English 1st patch, NSIS still updates our $LANGUAGE var to "Installer Language" (this is what the [!insertmacro MUI_LANGDLL_DISPLAY](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/591b00e538052543376e937a220aa6ef024ed10b/PowerEditor/installer/nppSetup.nsi#L190) should do after a user selected some specific language, but this time it fails completely somehow).
So any further dependence on the $LANGUAGE in our NSIS script is affected and we then also write that "Installer Language" to N++ lang-regkey for the future use (and not the 1033 value for English).

Reply:
> Yes and no.

Not really clear to me - Yes for what and No for what?

Reply:
OK thank you for your confirmation @xomx.
So for me the fix should only contain the part of putting English on the first position.
The other registry part should not be included in this PR.

Reply:
And what about the other possible `$LANGUAGE` dependencies like [this](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/591b00e538052543376e937a220aa6ef024ed10b/PowerEditor/installer/nsisInclude/mainSectionFuncs.nsh#L129)?